cirq/ops/common_gates_test.py

```diff
@@ -123,57 +123,62 @@ def test_z_init():
     assert cirq.Z**-1 == cirq.Z
 
 
-def test_z_control():
-    # Single qubit control on Z gives a CZ
-    assert cirq.Z.controlled() == cirq.CZ
-    assert cirq.Z.controlled(num_controls=1) == cirq.CZ
-    assert cirq.Z.controlled(control_values=((1,),)) == cirq.CZ
-    assert cirq.Z.controlled(control_qid_shape=(2,)) == cirq.CZ
-
-    # Also works for any ZPow.
-    assert cirq.ZPowGate(exponent=5).controlled() == cirq.CZPowGate(exponent=5)
+@pytest.mark.parametrize(
+    'input_gate, specialized_output',
+    [(cirq.Z, cirq.CZ), (cirq.CZ, cirq.CCZ), (cirq.X, cirq.CX),
+     (cirq.CX, cirq.CCX),
+     (cirq.ZPowGate(exponent=0.5), cirq.CZPowGate(exponent=0.5)),
+     (cirq.CZPowGate(exponent=0.5), cirq.CCZPowGate(exponent=0.5)),
+     (cirq.XPowGate(exponent=0.5), cirq.CNotPowGate(exponent=0.5)),
+     (cirq.CNotPowGate(exponent=0.5), cirq.CCXPowGate(exponent=0.5))])
+def test_specialized_control(input_gate, specialized_output):
```

Comment:
Is it worth checking that the unitaries are the same just to double check that the specialized gate actually is the same as the controlled input gate?

Reply:
Done. Thanks for the suggestion!